wsmux/session.go

```diff
@@ -0,0 +1,351 @@
+package wsmux
+
+import (
+	"bytes"
+	"encoding/binary"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+/*
+TODO: Add ping and pong handlers
+*/
+
+const (
+	defaultQueueSize            = 20
+	defaultStreamQueueSize      = 20
+	defaultKeepAliveInterval    = 30 * time.Second
+	defaultStreamAcceptDeadline = 30 * time.Second
+	deadCheckDuration           = 30 * time.Second
+)
+
+// Session implements net.Listener. Allows creating and acception multiplexed streams over ws
+type Session struct {
+	mu        sync.Mutex
+	streams   map[uint32]*stream
+	streamCh  chan *stream
+	conn      *websocket.Conn
+	acceptErr error
+
+	sendLock sync.Mutex
+
+	keepAliveInterval    time.Duration
+	streamAcceptDeadline time.Duration
+
+	logger Logger
+
+	nextID uint32
```

Comment:
Please add a comment... for all of these variables...
`nextID uint32 // next id for stream opened from this side (always incremented by 2)